rotkehlchen/chain/ethereum/adex/adex.py

```diff
@@ -713,117 +799,65 @@ def _get_identity_address_map(
-                        token_address=token_ethereum_address,
-                    )
-            except (KeyError, ValueError) as e:
-                msg = str(e)
-                log.error(
-                    'Error processing AdEx Tom pool unclaimed rewards. ',
-                    entry=entry,
-                    error=msg,
-                )
-                self.msg_aggregator.add_error(
-                    'Failed to deserialize the AdEx Tom pool unclaimed rewards. '
-                    'Check logs for details. Ignoring it.',
-                )
-                return {}
-
-        unclaimed_rewards = {}
-        for address in addresses:
-            total_adx_amount = sum(address_adx_amount.get(address, []))
-            total_dai_amount = sum(address_dai_amount.get(address, []))
-            # Discard addresses without amounts
-            if total_adx_amount + total_dai_amount > ZERO:
-                unclaimed_rewards[address] = UnclaimedReward(
-                    adx_amount=total_adx_amount / ADX_AMOUNT_MANTISSA,
-                    dai_amount=total_dai_amount / DAI_AMOUNT_MANTISSA,
-                )
-        return unclaimed_rewards
-
-    def _get_tom_pool_incentive(self) -> Optional[TomPoolIncentive]:
+    @staticmethod
+    def _get_tom_pool_incentive(fee_rewards: FeeRewards) -> TomPoolIncentive:
         """Get Tom pool incentive data (staking rewards).
 
         NB: the APR calculated here is the APY in the AdEx codebase and website stats.
         https://github.com/AdExNetwork/adex-staking/blob/master/src/actions/actions.js#L86
-
-        TODO: once AdEx rolls out issue #94, check how they handle incentives
-        with multiple channels, for instance how they process `periodEnd`.
-        https://github.com/AdExNetwork/adex-staking/issues/94
         """
-        fee_rewards = self._get_tom_pool_fee_rewards_from_api()
         total_staked_amount = ZERO
         total_reward_per_second = ZERO
         period_ends_at = Timestamp(0)
-        apr = ZERO
+        now = Timestamp(int(datetime.utcnow().timestamp()))
         for entry in fee_rewards:
             try:
-                if entry['channelArgs']['tokenAddr'] == A_ADX.ethereum_address:
-                    total_staked_amount += entry['stats']['currentTotalActiveStake']
-                    total_reward_per_second += entry['stats']['currentRewardPerSecond']
-                    # NB: `period_ends_at` is always overwritten
-                    period_ends_at = Timestamp(int(
-                        datetime.strptime(entry['periodEnd'], PERIOD_END_AT_FORMAT).timestamp(),
+                if (
+                    entry['channelArgs']['tokenAddr'] == A_ADX.ethereum_address and
+                    entry['channelId'] != TOM_POOL_FEE_REWARDS_ADX_LEGACY_CHANNEL
+                ):
+                    starts_at = Timestamp(int(
+                        datetime.strptime(entry['periodStart'], TOM_POOL_PERIOD_FORMAT).timestamp(),  # noqa: E501
```

Comment:
Use the already existing functions for these stuff: `create_timestamp` from utils/misc.py